### PR TITLE
Tests are rebuilt with cosm.js

### PR DESCRIPTION
**Pull request by ratik**

### CODE REVIEW

File: src/testcases/statom_short.test.ts [hunk 1]
```diff
@@ -172,6 +173,48 @@ describe('StAtom / Short', () => {
         ).toEqual(1_000);
       });
     });
+
+    describe('receive tokenized share', () => {
+      let tokenizedShare: Coin;
+      let startingStatomBalance: number;
+      let exchangeRate: number;
```
Comment (oopcode) [inline src/testcases/statom_short.test.ts:178]: looks like this is not used anywhere

File: src/testcases/statom_short.test.ts [hunk 2]
```diff
@@ -172,6 +173,48 @@ describe('StAtom / Short', () => {
...
+        const balances = await testState.wrapper.queryClient.bank.allBalances(
+          (
+            await testState.wallets.a.getAccounts()
+          )[0].address,
+        );
+        expect(
+          balances.find((b) => b.denom.startsWith('wasmvaloper')),
+        ).toBeFalsy();
+      });
+      test('statom balance must be increased', async () => {
```
Comment (oopcode) [inline src/testcases/statom_short.test.ts:209]: we would also like to check that after processing the tokenized share our delegation amount has increased

File: src/testcases/statom_short.test.ts (hunk 1 above)
Comment (ratik) [reply src/testcases/statom_short.test.ts:180]: fixed

File: src/testcases/statom_short.test.ts (hunk 2 above)
Comment (ratik) [reply src/testcases/statom_short.test.ts:216]: added both steps

**Review by swelf19 — COMMENTED** [review]
Amazing job was done.
But i think we missed one of the scenarios
1) The user unbonds some funds via vaucher of validator V1
2) V1 gets slashed
3) The user sends the vaucher(may be a part of vaucher) of the slashed validator to the hub

**Comment by foxpy** [comment]
I am probably doing something wrong, but `yarn test:short:statom` completely fails for me. By the way, I am using compiled contracts from `lido-cosmos-hub-conracts/main`.

**Comment by foxpy** [comment]
`yarn test:short:slashing` goes into infinite loop and keeps printing
```
  console.info
    execute wasm14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s0phg4d w/ {"bond_for_st_atom":{}} { amount: [ { amount: '250000', denom: 'stake' } ], gas: '1000000' }

      at src/helper/flow/execution.ts:50:11
```
and finally fails with a timeout. Is it an expected behaviour?

**Comment by ratik** [comment]
> yarn test:short:slashing

@foxpy please pull the changes (now slashing must happen much faster) and try again

**Comment by foxpy** [comment]
> try again

image

File: src/testcases/slashing.test.ts [hunk 3]
```diff
@@ -1,162 +1,129 @@
...
+
+  test('cycle of bonds', async () => {
+    for (let i = 0; i < ITER_TIMES; i++) {
+      const res = await testState.lasset.bond_for_statom(
+        testState.wallets.a,
+        2_000,
+      );
+      expect(res.code).toEqual(0);
+    }
+  });
```
Comment (foxpy) [inline src/testcases/slashing.test.ts:99]: Please add `2_000_000` timeout here. This test takes almost 700 seconds to run on my machine (default timeout is 600 seconds in this repo), so a timeout of 2000 seconds seems to be adequate even for very slow machines.

**Review by foxpy — CHANGES_REQUESTED** [review]

File: src/testcases/slashing.test.ts (hunk 3 above)
Comment (ratik) [reply src/testcases/slashing.test.ts]: done

File: src/testcases/statom_long.test.ts [hunk 4]
```diff
@@ -1,277 +1,180 @@
...
-  for (j = 0; j < 3; j++) {
-    for (i = 0; i < 25; i++) {
-      await testState.lasset.send_cw20_token(
-        statomContractAddress,
-        testState.wallets.b,
-        1_000_000,
-        { unbond: {} },
-        testState.lasset.contractInfo.lido_cosmos_hub.contractAddress
-      );
+  });
```
Comment (foxpy) [inline src/testcases/statom_long.test.ts:61]: I request setting here a timeout of `2_000_000` milliseconds.

File: src/testcases/statom_long.test.ts [hunk 5]
```diff
@@ -1,277 +1,180 @@
...
-  for (j = 0; j < 3; j++) {
-    for (i = 0; i < 25; i++) {
-      await testState.lasset.send_cw20_token(
-        statomContractAddress,
-        testState.wallets.c,
-        1_000_000,
-        { unbond: {} },
-        testState.lasset.contractInfo.lido_cosmos_hub.contractAddress
-      );
+  });
```
Comment (foxpy) [inline src/testcases/statom_long.test.ts:87]: I request setting here a timeout of `2_000_000` milliseconds.

File: src/testcases/statom_long.test.ts [hunk 6]
```diff
@@ -1,277 +1,180 @@
...
-  assert.ok(uatom_balance_b > initial_uatom_balance_b);
-  assert.ok(uatom_balance_c > initial_uatom_balance_c);
-
-  assert.ok(uatom_balance_lido_fee > initial_uatom_balance_lido_fee);
-}
-
-main()
-  .then(() => console.log("done"))
-  .catch(console.log);
+  });
```
Comment (foxpy) [inline src/testcases/statom_long.test.ts:112]: I request setting here a timeout of `2_000_000` milliseconds.

File: src/testcases/statom_long.test.ts [hunk 7]
```diff
@@ -1,277 +1,180 @@
...
+            statomContractAddress,
+            testState.wallets[wallet],
+            1_000,
+            { unbond: {} },
+            testState.lasset.contractInfo.lido_cosmos_hub.contractAddress,
+          );
+        }
+      }),
+    );
+  });
```
Comment (foxpy) [inline src/testcases/statom_long.test.ts:146]: I request setting here a timeout of `2_000_000` milliseconds.

**Review by foxpy — CHANGES_REQUESTED** [review]

**Comment by foxpy** [comment]
image

These tests are also quite slow, hence I am requesting a higher timeout.

**Review by oopcode — APPROVED** [review]

**Pull request merged**